ocdskingfisherprocess/web/app.py

```diff
@@ -51,11 +51,39 @@ def api_v1():
     return "OCDS Kingfisher APIs V1"
 
 
+@app.route("/api/v1/submit/end_collection_store/", methods=['POST'])
+def api_v1_submit_end_collection_store():
+    api_key = request.headers.get('Authorization', '')[len('ApiKey '):]
+    if not api_key or api_key not in config.web_api_keys:
+        return "ACCESS DENIED", 401
+
+    # TODO check all required fields are there!
+
+    database = DataBase(config=config)
+    store = Store(config=config, database=database)
+
+    collection_source = request.form.get('collection_source')
+    collection_data_version = parse_string_to_date_time(request.form.get('collection_data_version'))
+    collection_sample = parse_string_to_boolean(request.form.get('collection_sample', False))
+
+    store.load_collection(
+        collection_source,
+        collection_data_version,
+        collection_sample,
+    )
+
+    if store.is_collection_store_ended():
+        return "OCDS Kingfisher APIs V1 Submit - Already Done!"
+    else:
+        store.end_collection_store()
+        return "OCDS Kingfisher APIs V1 Submit"
+
+
 @app.route("/api/v1/submit/file/", methods=['POST'])
 def api_v1_submit_file():
-    api_key = request.headers.get('Authorization')[len('ApiKey '):]
+    api_key = request.headers.get('Authorization', '')[len('ApiKey '):]
     if not api_key or api_key not in config.web_api_keys:
-        return "ACCESS DENIED"  # TODO proper error
+        return "ACCESS DENIED", 401
 
     # TODO check all required fields are there!
```

Comment:
If the fields are not there will it mean the database will be in a state it was never intended to be in.

Reply:
Yes, that's a general clean up that should be applied to all end points tho. Issued to make sure it's done https://github.com/open-contracting/kingfisher-process/issues/47